flytepropeller/pkg/controller/nodes/task/handler.go

```diff
@@ -464,8 +500,19 @@ func (t Handler) invokePlugin(ctx context.Context, p pluginCore.Plugin, tCtx *ta
 		}
 	}
 
+	// Regardless of the observed phase, we always add the DeckUri to support real-time deck functionality.
+	// The deck should be accessible even if the task is still running or has failed.
+	// It's possible that the deck URI may not exist in remote storage yet or will never exist.
+	// So, it is console's responsibility to handle the case when the deck URI actually does not exist.
+	pluginTrns.AddDeckURI(ctx, tCtx)
+
 	switch pluginTrns.pInfo.Phase() {
 	case pluginCore.PhaseSuccess:
+		// This is to prevent the console from potentially checking the deck URI that does not exist if in final phase(PhaseSuccess).
+		err = pluginTrns.RemoveNonexistentDeckURI(ctx, tCtx)
```

Comment:
Does this do a `head` call on the deck URI for every task that succeeds? Two thoughts here:
(1) does the flyteadmin merge algorithm then remove the deckURI from the execution metadata?
(2) this is incurring a 20-30ms performance degredation to every task execution

Reply:
will take a look tmr, thank you!!!

Reply:
> Does this do a head call on the deck URI for every task that succeeds? 
yes it will do a `head` call

https://github.com/flyteorg/flyte/blob/b3330ba4430538f91ae9fc7d868a29a2e96db8bd/flyteplugins/go/tasks/pluginmachinery/ioutils/remote_file_output_reader.go#L306-L313

Reply:
how do you know the performance degradation?
did you use grafana or other performance tools?

Reply:
> does the flyteadmin merge algorithm then remove the deckURI from the execution metadata?

flyteadmin will set the `deckURI` in the `execution metadata` to `nil` if the propeller removes it.

Reply:
cc @hamersaw